components/region_cache_memory_engine/src/background.rs

```diff
@@ -754,29 +760,63 @@ impl Filter {
             default_cf_handle,
             write_cf_handle,
             mvcc_key_prefix: vec![],
-            cached_delete_key: None,
+            cached_mvcc_delete_key: None,
+            cached_skiplist_delete_key: None,
             remove_older: false,
             metrics: FilterMetrics::default(),
         }
     }
 
     fn filter(&mut self, key: &Bytes, value: &Bytes) -> Result<(), String> {
         self.metrics.total += 1;
-        let InternalKey { user_key, .. } = decode_key(key);
+        let InternalKey {
+            user_key, v_type, ..
+        } = decode_key(key);
 
         let (mvcc_key_prefix, commit_ts) = split_ts(user_key)?;
         if commit_ts > self.safe_point {
             return Ok(());
         }
 
+        // Just like what rocksdb compaction filter does, we does not hanlde internal
+        // keys(different MVCC versions of the same user key stands for different keys)
+        // that have been deleted as a tombstone. But we need to delete them anyway.
+        // They are below the safe point, so we can delete them directly now.
```

Comment:
```suggestion
        // Just like what rocksdb compaction filter does, we do not handle internal
        // keys (representing different MVCC versions of the same user key) that have
        // been marked as tombstones. However, these keys need to be deleted. 
        // Since they are below the safe point, we can safely delete them directly now.
```